src/safir/redis.py

```diff
@@ -0,0 +1,235 @@
+"""Redis database support."""
+
+from __future__ import annotations
+
+from collections.abc import AsyncIterator
+from typing import Generic, Optional, TypeVar
+
+import redis.asyncio as redis
+from cryptography.fernet import Fernet
+from pydantic import BaseModel
+
+from .slack.blockkit import SlackException, SlackMessage, SlackTextField
+
+__all__ = [
+    "PydanticRedisStorage",
+    "EncryptedPydanticRedisStorage",
+    "S",
```

Comment:
I don't think you want to export this. It's only used by mypy and thus is module local. (Unless this helps the API documentation look nicer, but it looks like you're still excluding it from Sphinx warnings, so presumably Sphinx doesn't yet understand TypeVars.)